primitives/asset/src/lib.rs

```diff
@@ -16,16 +15,42 @@
 // You should have received a copy of the GNU General Public License
 // along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
 
+//! Token Primitives
```

Comment:
How about moving the `asset` and `contract` to primitive s2s folder?

Reply:
But they are not only s2s used.

Reply:
O, Maybe part of them used in e2d. Is there any way to separate them?  Because it's hard to tell what is unique to s2s, what is unique to e2d, and what is common.

Reply:
Yes we separated them by different mode. see https://github.com/darwinia-network/darwinia-common/tree/master/primitives/contract/src/mapping_token_factory